File: 11 Container With Most Water/11 Container With Most Water.ts
function maxArea(height: number[]): number {
  let max: number = 0;
  let left: number = 0;
  let right: number = height.length - 1;

  while (left < right) {
    const area: number = Math.min(height[left], height[right]) * (right - left);
    max = Math.max(max, area);
    if (height[left] <= height[right]) {
      left++;
    } else {
      right--;
    }
  }

  return max;
}

File: 11 Container With Most Water/11 Container With Most Water.swift
class Solution {
  func maxArea(_ height: [Int]) -> Int {
    var maxWaterAmount: Int = 0
    var left: Int = 0
    var right: Int = height.count - 1

    while left < right {
      let width: Int = right - left
      let area: Int = min(height[left], height[right]) * width
      
      maxWaterAmount = max(maxWaterAmount, area)

      if height[left] <= height[right] {
        left += 1
      } else {
        right -= 1
      }
    }

    return maxWaterAmount  
  }
}